code/datums/outfits/outfit.dm

```diff
@@ -0,0 +1,216 @@
+	var/l_ear = null
+	var/r_ear = null
+	var/glasses = null	
+	var/l_pocket = null
+	var/r_pocket = null
+	var/suit_store = null
+
+	//The following vars must be paths
+	var/l_hand = null
+	var/r_hand = null
+	var/id = null
+	var/pda = null
+
+	var/internals_slot = null //ID of slot containing a gas tank
+	var/list/backpack_contents = list() //In the list(path=count,otherpath=count) format
+	var/list/implants = null //A list of implants that should be implanted
+
+/datum/outfit/proc/pre_equip(mob/living/carbon/human/H, visualsOnly = FALSE)
+	//to be overriden for customization depending on client prefs,species etc
+	return
+
+// Used to equip an item to the mob. Mainly to prevent copypasta for collect_not_del.
+/datum/outfit/proc/equip_item(mob/living/carbon/human/H, path, slot)
+	var/obj/item/I
+
+	if(islist(path))	//If its a list, select a random item
+		var/itempath = pick(path)
+		I = new itempath(H)
+	else if(!isnum(path) && gear_datums[path]) //If its something else, we´ll check if its a gearpath and try to spawn it
+		var/datum/gear/G = gear_datums[path]
+		I = G.spawn_random()
+	else
+		I = new path(H) //As fallback treat it as a path
+
+	if(collect_not_del)
+		H.equip_or_collect(I, slot)
+	else
+		H.equip_to_slot_or_del(I, slot)
+
+/datum/outfit/proc/post_equip(mob/living/carbon/human/H, visualsOnly = FALSE)
+	//to be overriden for toggling internals
+	imprint_idcard_pda(H)
+	return
+
+/datum/outfit/proc/equip(mob/living/carbon/human/H, visualsOnly = FALSE)
+	pre_equip(H, visualsOnly)
+
+	//Start with uniform,suit,backpack for additional slots
+	if(back)
+		equip_item(H, back, slot_back)
+	if(uniform)
+		equip_item(H, uniform, slot_w_uniform)
+	if(suit)
+		equip_item(H, suit, slot_wear_suit)
+	if(belt)
+		equip_item(H, belt, slot_belt)
+	if(gloves)
+		equip_item(H, gloves, slot_gloves)
+	if(shoes)
+		equip_item(H, shoes, slot_shoes)
+	if(head)
+		equip_item(H, head, slot_head)
+	if(mask)
+		equip_item(H, mask, slot_wear_mask)
+	if(l_ear)
+		equip_item(H, l_ear, slot_l_ear)
+	if(r_ear)
+		equip_item(H, r_ear, slot_r_ear)
+	if(glasses)
+		equip_item(H, glasses, slot_glasses)
+
+	if(suit_store)
+		equip_item(H, suit_store, slot_s_store)
+
+	if(l_hand)
+		H.put_in_l_hand(new l_hand(H))
+	if(r_hand)
+		H.put_in_r_hand(new r_hand(H))
+
+	if(pda)
+		var/obj/item/I = new pda(H)
+		H.equip_or_collect(I, slot_wear_id)
+
+	if(id)
+		var/obj/item/device/pda/P = H.wear_id
+		var/obj/item/I = new id(H)
+		if(istype(P))
+			I.forceMove(P)
+			P.id = I
+		else
+			H.equip_or_collect(I, slot_wear_id)
+
+
+	if(!visualsOnly) // Items in pockets or backpack don't show up on mob's icon.
+		if(l_pocket)
+			equip_item(H, l_pocket, slot_l_store)
+		if(r_pocket)
+			equip_item(H, r_pocket, slot_r_store)
+
+		for(var/path in backpack_contents)
+			var/number = backpack_contents[path]
+			for(var/i in 1 to number)
+				H.equip_or_collect(new path(H), slot_in_backpack)
+
+	post_equip(H, visualsOnly)
+
+	if(!visualsOnly)
+		apply_fingerprints(H)
+
+	if(implants)
```

Comment:
Shouldn't this be guarded by `visualsOnly` as well? Considering implants have no visual effect.